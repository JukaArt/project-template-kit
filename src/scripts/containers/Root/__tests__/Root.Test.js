import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import createHistory from 'history/createBrowserHistory';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Root from 'Containers/Root';
import AppRouter from 'Containers/AppRouter';

describe('Connected Root', () => {
  const mockStore = configureMockStore();
  let root, store, history, initialItems;

  beforeEach( () => {
    initialItems = ['one'];
		const initialState = {
      items: initialItems,
      num: 0,
		}
    store = mockStore(initialState);
    history = createHistory();
  })

  describe('state provided by the store', () => {
		beforeEach( () => {
      root = shallow(<Provider store={store} history={history}><Root/></Provider>);
    });

    it('should exist', () => {
      expect(root.exists()).toEqual(true);
    });

    it('Should be a structure', () => {
      expect(toJson(root)).toMatchSnapshot();
    })
  })

})
