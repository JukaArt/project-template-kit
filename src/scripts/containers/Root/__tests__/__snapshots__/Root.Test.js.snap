// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Root state provided by the store Should be a structure 1`] = `
<Component
  history={Object {}}
  store={Object {}}
/>
`;
